server/webapp/WEB-INF/rails.new/app/presenters/api_v1/config/materials/git_material_representer.rb

```diff
@@ -27,6 +27,7 @@ class GitMaterialRepresenter < ScmMaterialRepresenter
                    value.blank? ? self.setBranch('master') : self.setBranch(value)
                  }
         property :submodule_folder
+        property :shallow_clone
```

Comment:
You will also need to submit a PR to https://github.com/gocd/api.go.cd to document this property, and clearly indicate that it was introduced in 16.3.0. See `agent_state` or `build_state` attribute at https://api.go.cd/current/#the-agent-object for an example.